Makefile

```diff
@@ -116,19 +125,28 @@ mock-gen:
 	mockgen -source=cmd/tools/jwt/app/configmap_updater.go -destination ./cmd/tools/jwt/app/mock_app/configmap_updater.go
 	mockgen -source=cmd/tools/jwt/app/kubernetes.go -destination ./cmd/tools/jwt/app/mock_app/kubernetes.go
 
-# find or download controller-gen
-# download controller-gen if necessary
-controller-gen:
-ifeq (, $(shell which controller-gen))
-	@{ \
-	set -e ;\
-	CONTROLLER_GEN_TMP_DIR=$$(mktemp -d) ;\
-	cd $$CONTROLLER_GEN_TMP_DIR ;\
-	go mod init tmp ;\
-	go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.6.2 ;\
-	rm -rf $$CONTROLLER_GEN_TMP_DIR ;\
-	}
-CONTROLLER_GEN=$(GOBIN)/controller-gen
-else
-CONTROLLER_GEN=$(shell which controller-gen)
-endif
+CONTROLLER_GEN = $(shell pwd)/bin/controller-gen
```

Comment:
why download it to the current directory?

Reply:
Ensure that we won't break the existing binary with different verison.

Reply:
The directory `bin` is temporary. We need to download it again once it was deleted.

Reply:
It will automatically download towards bin folder. Please see:

https://github.com/kubesphere/ks-devops/blob/51bd7fbc9ce4afeed32a1a490e8e6193ea6486bb/Makefile#L142-L152

Reply:
User can set GOPROXY environment for speeding up download. So I don't think it is a problem.

Reply:
It's not necessary to download it if contrller-gen existed. And please don't forget `bin` is just a temporary directory. Put it into a temporary is not a mutli-version solution.

Reply:
Please see:
 https://github.com/kubesphere/ks-devops/blob/51bd7fbc9ce4afeed32a1a490e8e6193ea6486bb/Makefile#L143

If a binary exists in `bin` folder, the `go-get-tool` won't download it again.

Reply:
What if I clone this project in different serval directories of my computer, why should I download all these tools many times?
What if I already have `controller-gen` in my PATH, why should I download it again?

Reply:
How do we know what's your version of controller-gen or other tools. We have to guarrantee we use controller-gen or other tools with specific version.

BTW,the scenary that users clone repo in different places is unusual.

Reply:
As I said before `bin` is a temporary directory. So we can delete it at any time. But these tools are not temporary.

About the version, your solution cannot guarantee it too.

Reply:
So what's happen if you delete the temporary folder `bin` here? I didn't get your point. It's just like dependencies which we can delete at any time.

Reply:
I don't want to download these tools everytime when I delete the `bin` directory.

Reply:
I'll stick my own view. But for this PR, I will revert the Makefile for stopping controversial discussion.